gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -95,6 +100,8 @@ public class TrinoQueryProperties
     private Set<String> catalogs = ImmutableSet.of();
     private Set<String> schemas = ImmutableSet.of();
     private Set<String> catalogSchemas = ImmutableSet.of();
+    private Optional<Call> procedure = Optional.empty();
+    private List<CallArgument> procedureArguments = ImmutableList.of();
```

Comment:
Please consider changing these fields to `Optional<String>` and `List<SerializableCallArgument>`.

Reply:
`ProxyUtils::extractQueryIdFromRequest` checked that `argument instanceof StringLiteral`, so when changing `procedureArguments` to `List<SerializableCallArgument>` I added a field to record the original type. Is there a better way to go about this?